v2/internal/controllers/latest_reconciled_generation_test.go

```diff
@@ -65,7 +67,7 @@ func Test_Latest_Reconciled_Generation_Reconciles_AllEvents(t *testing.T) {
 	old := agentPool.DeepCopy()
 	agentPool.Spec.OrchestratorVersion = to.Ptr("1.27.1")
 
-	tc.Patch(old, agentPool)
+	tc.PatchResourceAndWaitForState(old, agentPool, metav1.ConditionFalse, conditions.ConditionSeverityInfo)
```

Comment:
Did you confirm that doing this was still able to trigger the test on the behavior we wanted (that 2 PUTs happen before the first reconcile is finished)?

Reply:
Yep, It does take that - generation mismatch path on re-runs.